modules/tensor_mechanics/src/actions/TensorMechanicsAction.C

```diff
@@ -451,6 +470,151 @@ TensorMechanicsAction::actOutputGeneration()
+      {
+        remove_list = mat_params.get<std::vector<std::string>>("reg_props_in");
+        add_list = mat_params.get<std::vector<std::string>>("ad_props_out");
+      }
+      if (mat_params.isParamValid("ad_props_in") && mat_params.isParamValid("reg_props_out"))
+      {
+        remove_list = mat_params.get<std::vector<std::string>>("ad_props_in");
+        add_list = mat_params.get<std::vector<std::string>>("reg_props_out");
+      }
+
+      // These vectors are the same size as checked in MaterialConverter
+      for (unsigned int index = 0; index < remove_list.size(); index++)
+        remove_add_map.emplace(remove_list[index], add_list[index]);
+    }
+  }
+  // All the materials have been accounted for, now remove or add parts
+
+  // Correlate BlockIDs with eigenstrain names (remove irrelevant subdomainIDs)
+  if (!_subdomain_ids.empty())
+    for (auto i : material_eigenstrain_map)
+      if (i.second != _subdomain_ids)
+        eigenstrain_set.erase(i.first);
+
+  // Remove names which aren't eigenstrains (converter properties)
+  for (auto remove_add_index : remove_add_map)
+  {
+    const bool is_in = eigenstrain_set.find(remove_add_index.first) != eigenstrain_set.end();
+    if (is_in)
+    {
+      eigenstrain_set.erase(remove_add_index.first);
+      eigenstrain_set.insert(remove_add_index.second);
+    }
+  }
+
+  for (auto index : remove_reduced_set)
+  {
+    if (eigenstrain_set.find(index) != eigenstrain_set.end())
+      eigenstrain_set.erase(index);
+  }
+
+  // Compare the blockIDs set of eigenstrain names with the vector of _eigenstrain_names for the
+  // current subdomainID
+  std::set_union(eigenstrain_set.begin(),
+                 eigenstrain_set.end(),
+                 eigenstrain_names_copy.begin(),
+                 eigenstrain_names_copy.end(),
+                 std::inserter(verified_eigenstrain_names, verified_eigenstrain_names.begin()));
+
+  // Ensure the eigenstrain names previously passed include any missing names
+  _eigenstrain_names.clear();
+  _eigenstrain_names.resize(verified_eigenstrain_names.size());
+  std::copy(verified_eigenstrain_names.begin(),
+            verified_eigenstrain_names.end(),
+            _eigenstrain_names.begin());
+
+  // Since the automatic eigenstrain names have been collected, return the list
+  auto list_subdomains = _problem->mesh().meshSubdomains();
+  // If no defined blocks
+  if (list_subdomains.size() == 1 && *list_subdomains.begin() == 0)
+    _console << COLOR_CYAN << "*** Automatic Eigenstrain Names ***" << COLOR_DEFAULT << std::endl;
+  // Only print once for multiple blocks
+  else
+    for (auto currentNum : _subdomain_ids)
+      if (currentNum == *list_subdomains.begin())
+        _console << COLOR_CYAN << "*** Automatic Eigenstrain Names ***" << COLOR_DEFAULT
+                 << std::endl;
+  _console << _name << ": " << Moose::stringify(_eigenstrain_names) << std::endl;
+}
```

Comment:
The for loop is used to emulate the output a mooseInfo string would follow.  Since the TMA is looped over how every many blocks are within the TMA, the 'Automatic Eigenstrain Names' get printed.  Running the `expect_out` test condition would have to include this statement as a result

Reply:
![Screen Shot 2020-10-21 at 2 42 14 PM](https://user-images.githubusercontent.com/33581084/96785482-14d8fe80-13ac-11eb-9920-2f89401b77b2.png)
![Screen Shot 2020-10-21 at 2 44 50 PM](https://user-images.githubusercontent.com/33581084/96785492-17d3ef00-13ac-11eb-8fe2-29bee2683a15.png)

Reply:
> result

_console is the issue.  This doesn't occur with Moose::out